plugin/src/main/java/io/fabric8/maven/plugin/SetupMojo.java

```diff
@@ -0,0 +1,281 @@
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+        if (createOrUpdateFabric8MavenPlugin(doc)) {
+            if (generateBackupPoms) {
+                try {
+                    pom.renameTo(backupPomFileName);
+                    getLog().info("Created backup pom.xml file at: " + backupPomFileName);
+                } catch (Exception e) {
+                    throw new MojoExecutionException("Failed to create backup: " + backupPomFileName + ". " + e, e);
+                }
+            }
+            getLog().info("Updating the pom " + pom);
+            try {
+                DomHelper.save(doc, pom);
+            } catch (Exception e) {
+                getLog().error("Failed to update pom " + pom + ". " + e, e);
+                throw new MojoExecutionException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private boolean createOrUpdateFabric8MavenPlugin(Document doc) throws MojoExecutionException {
+        boolean updated = false;
+        String latestVersion;
+        try {
+            latestVersion = MavenUtil.getVersion(PLUGIN_GROUPID, PLUGIN_ARTIFACTID);
+        } catch (IOException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+        // lets check that there's a property defined
+        String currentVersion = project.getProperties().getProperty(FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
+        String versionExpression = "${" + FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY + "}";
+        Element fmpPlugin = findPlugin(doc, PLUGIN_GROUPID, PLUGIN_ARTIFACTID);
+        boolean pluginMissing = fmpPlugin == null;
+        if (updateVersion || pluginMissing) {
+            if (useVersionProperty) {
+                Element documentElement = doc.getDocumentElement();
+                Element properties = firstChild(documentElement, "properties");
+                if (properties == null) {
+                    properties = addChildAfter(appendAfterLastElement(properties, doc.createTextNode("\n      ")), "properties");
+                }
+                if (Strings.isNullOrBlank(currentVersion)) {
+                    addChildAfter(appendAfterLastElement(properties, doc.createTextNode("\n    ")), FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY, latestVersion);
+                    updated = true;
+                } else if (!Objects.equals(currentVersion, latestVersion)) {
+                    Element propertyElement = DomHelper.firstChild(properties, FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
+                    if (propertyElement != null) {
+                        propertyElement.setTextContent(latestVersion);
+                        updated = true;
+                    }
+                }
+            } else {
+                versionExpression = latestVersion;
+            }
+        }
+
+        if (pluginMissing) {
+            fmpPlugin = findOrAddPlugin(doc, PLUGIN_GROUPID, PLUGIN_ARTIFACTID, versionExpression);
+            updated = true;
```

Comment:
the purpose of the flag is to track if it was missing - i.e. if we should add the executions. If we set it to false then we'll never ever add the executions. I added this logic based on your previous comment about only adding executions when we first add the plugin